bugshaker/src/main/java/com/github/stkent/bugshaker/BugShaker.java

```diff
@@ -44,277 +44,260 @@
-                    = (SensorManager) application.getSystemService(SENSOR_SERVICE);
-            final ShakeDetector shakeDetector = new ShakeDetector(this);
-
-            final boolean didStart = shakeDetector.start(sensorManager);
-
-            if (didStart) {
-                logger.d("Shake detection successfully started!");
-            } else {
-                logger.e("Error starting shake detection: hardware does not support detection.");
-            }
-        } else {
-            logger.e("Error starting shake detection: device cannot send emails.");
-        }
-
-        startAttempted = true;
-    }
-
-    @Override
-    public void hearShake() {
-        logger.d("Shake detected!");
-
-        feedbackEmailFlowManager.startFlowIfNeeded(
-                emailAddresses,
-                emailSubjectLine,
-                ignoreFlagSecure);
-    }
-
-    /**
-     * @return a MapScreenshotProvider if the embedding application utilizes the Google Maps Android
-     *         API, and a BasicScreenshotProvider otherwise
-     */
-    private ScreenshotProvider getScreenshotProvider() {
-        try {
-            Class.forName(
-                    "com.google.android.gms.maps.GoogleMap",
-                    false,
-                    BugShaker.class.getClassLoader());
-
-            logger.d("Detected that embedding app includes Google Maps as a dependency.");
-
-            return new MapScreenshotProvider(application, logger);
-        } catch (final ClassNotFoundException e) {
-            logger.d("Detected that embedding app does not include Google Maps as a dependency.");
-
-            return new BasicScreenShotProvider(application, logger);
-        }
-    }
-
-    private DialogProvider getAlertDialogProvider() {
-        if (alertDialogType == AlertDialogType.APP_COMPAT) {
-            try {
-                Class.forName(
-                        "android.support.v7.app.AlertDialog",
-                        false,
-                        BugShaker.class.getClassLoader());
-
-                logger.d("Using AppCompat dialogs as requested.");
-
-                return new AppCompatDialogProvider();
-            } catch (final ClassNotFoundException e) {
-                logger.e("AppCompat dialogs requested, but class not found.");
-                logger.e("Falling back to native dialogs.");
-
-                return new NativeDialogProvider();
-            }
-        } else {
-            logger.d("Using native dialogs as requested.");
-
-            return new NativeDialogProvider();
-        }
-    }
+	private static final String RECONFIGURATION_EXCEPTION_MESSAGE =
+		"Configuration must be completed before calling assemble or start";
+
+	private static BugShaker sharedInstance;
+
+	public static final String PREFS_NAME = "name";
+	public static final String PREFS_NAME2 = "name2";
+
+	public static final String PREFS_KEY = "key";
+	public static final String PREFS_KEY2 = "key2";
```

Comment:
We should be more descriptive about these variables. Another developer should know at first look what these variables do.